exdir/core/group.py

```diff
@@ -113,12 +113,15 @@ def create_dataset(self, name, shape=None, dtype=None,
 
         prepared_data, attrs, meta = ds._prepare_write(data, self.plugin_manager.dataset_plugins.write_order)
 
+        if not isinstance(prepared_data, np.ndarray) and prepared_data is not None:
+            prepared_data = np.array(prepared_data)
```

Comment:
Which types will be affected by this? I remember something about quantities automatically getting converted the wrong way (i.e. not by the plugin if there was no plugin) because there was some support for conversion that we didn't expect. If this only affects lists (and other types that you'd really expect to get converted to an np.array), I'm all for it. But if this may trigger some unexpected conversion that a plugin should have done when the plugin is not enabled, I think we should only add it for types we already know (such as lists).